setup.ps1

```diff
@@ -0,0 +1,339 @@
+    Write-Log "Virtual environment not found. Creating virtual environment..." -Level 'ERROR'
+    & $pythonExecutable -m venv "$VenvName"
+    if (-Not (Test-Path $VenvPython)) {
+      Write-Log "Failed to create virtual environment." -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+    Write-Log "Virtual environment created successfully."
+  }
+
+  # Activate the virtual environment
+  $Global:OldVirtualPath = $env:PATH
+  $env:PATH = "$VenvDir\Scripts;$env:PATH"
+
+  # Pull latest updates
+  Write-Log "Pulling latest updates..."
+  & "C:\Program Files\Git\cmd\git.exe" pull | Out-File $LogFilePath -Append 2>&1
+  if ($LASTEXITCODE -ne 0) {
+    Write-Log "Failed to pull updates from Git." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+
+  if (-not (Test-Path "$VenvDir\Lib\site-packages\talib")) {
+    # Install TA-Lib using the virtual environment's pip
+    Write-Log "Installing TA-Lib using virtual environment's pip..."
+    & $VenvPython -m pip install --find-links=build_helpers\ --prefer-binary TA-Lib | Out-File $LogFilePath -Append 2>&1
+  }
+
+  # Present options for requirement files
+  $selectedIndices = Get-UserSelection -prompt "Select which requirement files to install:" -options $RequirementFiles -defaultChoice 'A'
+
+  # Cache the selected requirement files
+  $selectedRequirementFiles = @()
+  foreach ($index in $selectedIndices) {
+    if ($index -lt 0 -or $index -ge $RequirementFiles.Length) {
+      Write-Log "Invalid selection. Please try again using the allowed letters." -Level 'ERROR'
+      continue
+    }
+  
+    $filePath = Join-Path $PSScriptRoot $RequirementFiles[$index]
+    if (Test-Path $filePath) {
+      $selectedRequirementFiles += $filePath
+    }
+    else {
+      Write-Log "Requirement file not found: $filePath" -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+  }
+
+  # Merge the contents of the selected requirement files
+  if ($selectedRequirementFiles.Count -gt 0) {
+    Write-Log "Merging selected requirement files..."
+    $mergedDependencies = @()
+    foreach ($file in $selectedRequirementFiles) {
+      $mergedDependencies += Get-Requirements $file
+    }
+    # Avoid duplicate dependencies
+    $mergedDependencies = $mergedDependencies | Select-Object -Unique
+
+    # Create a temporary directory
+    $tempDir = Join-Path $env:TEMP "freqtrade_requirements"
+    New-Item -ItemType Directory -Path $tempDir -Force | Out-Null
+
+    # Create a temporary file with the merged dependencies
+    $tempFile = Join-Path $tempDir "requirements.txt"
+    $mergedDependencies | Out-File $tempFile
+
+    # Install the merged dependencies
+    Write-Log "Installing merged dependencies..."
+    & $VenvPython -m pip install -r $tempFile
+    if ($LASTEXITCODE -ne 0) {
+      Write-Log "Failed to install merged dependencies. Exiting now..." -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+
+    # Remove the temporary directory
+    Remove-Item $tempDir -Recurse -Force
+  }
```

Comment:
This whole section looks a lot more complex than it needs to be, and does things that are not necessary.

pip can handle multiple requests files just fine (`pip install -r requirements1.txt -r requirements2.txt  ...`) - there's no need to try and manually combine these files.

Reply:
Just increased speed, but okay I undo it

Reply:
well i don't see how it increases speed.
What i'm saying is still 1 call to pip - so **writing** a new requirements file will be slower - even if it's just slightly slower.

Reply:
I wanna know it exactly now. I'll do a performance benchmark. I think it is faster because pip doesn't need to check a package multiple times and writing to a file is done pretty fast compared to all the checks pip needs to do. But we'll see.